client/rest-high-level/src/main/java/org/elasticsearch/client/IndicesClient.java

```diff
@@ -58,6 +61,31 @@ public DeleteIndexResponse deleteIndex(DeleteIndexRequest deleteIndexRequest, He
      */
     public void deleteIndexAsync(DeleteIndexRequest deleteIndexRequest, ActionListener<DeleteIndexResponse> listener, Header... headers) {
         restHighLevelClient.performRequestAsyncAndParseEntity(deleteIndexRequest, Request::deleteIndex, DeleteIndexResponse::fromXContent,
-            listener, Collections.emptySet(), headers);
+            listener, emptySet(), headers);
+    }
+
+    public IndicesExistsResponse exists(IndicesExistsRequest indicesExistsRequest, Header... headers) throws IOException {
+        return restHighLevelClient.performRequest(
+            indicesExistsRequest,
+            Request::indicesExist,
+            this::parseIndicesExistResp,
+            emptySet(),
+            headers
+        );
+    }
+
+    private IndicesExistsResponse parseIndicesExistResp(Response response) {
```

Comment:
It might have been better to move this method into `IndicesExistsResponse`, but `Response` is not visible from `IndicesExistsResponse`.

Reply:
use the already existing convertExistsResponse from RestHighLevelClient, you can see how this is done in the ping method for instance.